src/commonMain/kotlin/exchange.dydx.abacus/output/input/AdjustIsolatedMarginInput.kt

```diff
@@ -0,0 +1,175 @@
+
+                return if (existing?.needsSize != needsSize
+                ) {
+                    AdjustIsolatedMarginInputOptions(needsSize)
+                } else {
+                    existing
+                }
+            }
+            Logger.d { "Adjust Isolated Margin Input Options not valid" }
+            return null
+        }
+    }
+}
+
+@JsExport
+@Serializable
+data class AdjustIsolatedMarginInputSummary(
+    val crossFreeCollateral: Double?,
+    val crossMarginUsage: Double?,
+    val positionMargin: Double?,
+    val positionLeverage: Double?,
+    val liquidationPrice: Double?,
+) {
+    companion object {
+        internal fun create(
+            existing: AdjustIsolatedMarginInputSummary?,
+            parser: ParserProtocol,
+            data: Map<*, *>?
+        ): AdjustIsolatedMarginInputSummary? {
+            Logger.d { "creating Adjust Isolated Margin Input Summary\n" }
+
+            data?.let {
+                val crossFreeCollateral = parser.asDouble(data["crossFreeCollateral"])
+                val crossMarginUsage = parser.asDouble(data["crossMarginUsage"])
+                val positionMargin = parser.asDouble(data["positionMargin"])
+                val positionLeverage = parser.asDouble(data["positionLeverage"])
+                val liquidationPrice = parser.asDouble(data["liquidationPrice"])
+
+                return if (
+                    existing?.crossFreeCollateral != crossFreeCollateral ||
+                    existing?.crossMarginUsage != crossMarginUsage ||
+                    existing?.positionMargin != positionMargin ||
+                    existing?.positionLeverage != positionLeverage ||
+                    existing?.liquidationPrice != liquidationPrice
+                ) {
+                    AdjustIsolatedMarginInputSummary(
+                        crossFreeCollateral,
+                        crossMarginUsage,
+                        positionMargin,
+                        positionLeverage,
+                        liquidationPrice,
+                    )
+                } else {
+                    existing
+                }
+            }
+            Logger.d { "Adjust Isolated Margin Input Summary not valid" }
+            return null
+        }
+    }
+}
+
+@JsExport
+@Serializable
+enum class IsolatedMarginAdjustmentType(val rawValue: String) {
+    Add("ADD"),
+    Remove("REMOVE");
+
+    companion object {
+        operator fun invoke(rawValue: String) =
+            IsolatedMarginAdjustmentType.values().firstOrNull { it.rawValue == rawValue }
+    }
+}
+
+@JsExport
+@Serializable
+data class AdjustIsolatedMarginInput(
+    val type: IsolatedMarginAdjustmentType,
+    val amount: String?,
+    val childSubaccountNumber: Int?,
+    val adjustIsolatedMarginInputOptions: AdjustIsolatedMarginInputOptions?,
+    val summary: AdjustIsolatedMarginInputSummary?,
+    val errors: String?,
+    val errorMessage: String?,
+    val fee: Double?,
```

Comment:
same here, are there required fields we can make non-nullable?

Reply:
I think the only non-nullable field that we want would be `type` which is already the case.